galley/pkg/config/processor/transforms/authpolicy/create.go

```diff
@@ -21,26 +21,44 @@ import (
 
 	"istio.io/istio/galley/pkg/config/collection"
 	"istio.io/istio/galley/pkg/config/event"
+	"istio.io/istio/galley/pkg/config/processing"
 	"istio.io/istio/galley/pkg/config/processor/metadata"
+	"istio.io/istio/galley/pkg/config/processor/transforms"
 	"istio.io/istio/galley/pkg/config/scope"
 )
 
-// Create a new Direct transformer.
-func Create() []event.Transformer {
-	return []event.Transformer{
-		event.NewFnTransform(
-			collection.Names{metadata.K8SAuthenticationIstioIoV1Alpha1Policies},
-			collection.Names{metadata.IstioAuthenticationV1Alpha1Policies},
-			nil,
-			nil,
-			handler(metadata.IstioAuthenticationV1Alpha1Policies),
+func GetInfo() transforms.Infos {
+	policiesInput := collection.Names{metadata.K8SAuthenticationIstioIoV1Alpha1Policies}
+	policiesOutput := collection.Names{metadata.IstioAuthenticationV1Alpha1Policies}
+	meshPoliciesInput := collection.Names{metadata.K8SAuthenticationIstioIoV1Alpha1Meshpolicies}
+	meshPoliciesOutput := collection.Names{metadata.IstioAuthenticationV1Alpha1Meshpolicies}
+
+	return []*transforms.Info{
+		transforms.NewInfo(
```

Comment:
I'd recommend creating a helper to simplify this code. It can look like this:

```golang
func init() {
   // This internally calls event.NewFnTransform()
    info := transforms.InfoForSimpleTransform(
       collection.Names{metadata.K8SAuthenticationIstioIoV1Alpha1Policies},
       collection.Names{metadata.IstioAuthenticationV1Alpha1Policies},
      handler(metadata. IstioAuthenticationV1Alpha1Policies))

   transforms.Register(info)
}
```

events.NewFnTransform is defined, so that it is easy to create transforms. I think that model can carry over to here as well.

Reply:
Or, alternatively, you can simplify, by making use of the fact that for some set of transformers, the processoroptions is not needed:

```golang
func init() {
   // Here, the transform is created eagerly, allowing the InfoFromTransform to query Inputs/Outputs and create
  // a "factory" method that simply returns the passed-in transform.
    info := transforms.InfoFromTransform(handler(metadata. IstioAuthenticationV1Alpha1Policies))

 transforms.Register(info)
}
```

Reply:
Added a helper for the basic cases.